doc/src/challenges/0023-vec-pt1.md

```diff
@@ -0,0 +1,81 @@
+# Challenge 17: Verify the safety of `Vec` functions part 1
```

Comment:
This PR is missing an update to doc/src/SUMMARY.md to link to this new challenge.